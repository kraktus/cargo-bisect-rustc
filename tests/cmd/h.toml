bin.name = "cargo-bisect-rustc"
args = "-h"
