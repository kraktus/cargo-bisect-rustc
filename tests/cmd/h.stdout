cargo-bisect-rustc 0.6.3
Bisects rustc toolchains with rustup

USAGE:
    cargo-bisect-rustc [OPTIONS] [-- <COMMAND_ARGS>...]

ARGS:
    <COMMAND_ARGS>...    Arguments to pass to cargo or the file specified by --script during
                         tests

OPTIONS:
    -a, --alt                       Download the alt build instead of normal build
        --access <ACCESS>           How to access Rust git repository [github|checkout]
        --by-commit                 Bisect via commit artifacts
    -c, --component <COMPONENTS>    additional components to install
        --end <END>                 Right bound for search (*with* regression). You can use a date
                                    (YYYY-MM-DD), git tag name (e.g. 1.58.0) or git commit SHA.
        --force-install             Force installation over existing artifacts
    -h, --help                      Print help information
        --host <HOST>               Host triple for the compiler [default: unknown]
        --install <INSTALL>         Install the given artifact
        --preserve                  Preserve the downloaded artifacts
        --preserve-target           Preserve the target directory used for builds
        --prompt                    Manually evaluate for regression with prompts
        --regress <REGRESS>         Custom regression definition [default: error]
        --script <SCRIPT>           Script replacement for `cargo build` command
        --start <START>             Left bound for search (*without* regression). You can use a date
                                    (YYYY-MM-DD), git tag name (e.g. 1.58.0) or git commit SHA.
    -t, --timeout <TIMEOUT>         Assume failure after specified number of seconds (for bisecting
                                    hangs)
        --target <TARGET>           Cross-compilation target platform
        --test-dir <TEST_DIR>       Root directory for tests [default: .]
    -V, --version                   Print version information
        --verbose                   
        --with-dev                  Download rustc-dev [default: no download]
        --with-src                  Download rust-src [default: no download]
        --without-cargo             Do not install cargo [default: install cargo]

EXAMPLES:
    Run a fully automatic nightly bisect doing `cargo check`:
    ```
    cargo bisect-rustc --start 2018-07-07 --end 2018-07-30 --test-dir ../my_project/ -- check
    ```

    Run a PR-based bisect with manual prompts after each run doing `cargo build`:
    ```
    cargo bisect-rustc --start 6a1c0637ce44aeea6c60527f4c0e7fb33f2bcd0d /
      --end 866a713258915e6cbb212d135f751a6a8c9e1c0a --test-dir ../my_project/ --prompt -- build
    ```
